tests/artifactorysecurity_test.go

```diff
@@ -0,0 +1,102 @@
+package tests
+
+import (
+	"github.com/jfrog/jfrog-client-go/artifactory/services"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"testing"
+)
+
+// Teardown should revoke these tokens
+var tokensToRevoke []string
+
+func TestToken(t *testing.T) {
+	t.Run("CreateToken", createTokenTest)
+	t.Run("RevokeToken", revokeTokenTest)
+	t.Run("RefreshToken", refreshTokenTest)
+	t.Run("GetTokens", getTokensTest)
+	teardown()
+}
+
+func createTokenTest(t *testing.T) {
+	token, err := createToken()
+	if err != nil {
+		t.Error(err)
+	}
+	if token.AccessToken == "" {
+		t.Error("Failed to create access token")
+	}
+	tokensToRevoke = append(tokensToRevoke, token.RefreshToken)
+}
+
+func revokeTokenTest(t *testing.T)  {
+	token, err := createToken()
+	if err != nil {
+		t.Error(err)
+	}
+	err = revokeToken(token.RefreshToken)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func refreshTokenTest(t *testing.T) {
+	token, err := createToken()
+	if err != nil {
+		t.Error(err)
+	}
+	params := services.NewRefreshTokenParams()
+	params.RefreshToken = token.RefreshToken
+	params.AccessToken = token.AccessToken
+	newToken, err := testsSecurityService.RefreshToken(params)
+	if err != nil {
+		t.Error(err)
+	}
+	tokensToRevoke = append(tokensToRevoke, newToken.RefreshToken)
+}
+
+func getTokensTest(t *testing.T) {
```

Comment:
This test ~doubles the runtime of the test suite in my experience, due to the performance of the artifactory endpoint. I'm not sure if it's going to be a welcome addition or if it should be removed.